### ISSUE TITLE
Missing commodity caused server 500

### ISSUE THREAD

**Issue by xiaoruoruo** [aumayr/fava#338]
```
option "operating_currency" "USD"

2016-01-01 open Expenses:Food
2016-01-01 open Assets:Cash

2016-06-05 * "Test"
    Expenses:Food 100
    Assets:Cash
```

[E 160702 23:35:59 wsgi:355] 500 GET /beancount/income_statement/ (::1) 27.91ms

Server 500 is generic internal error, I had no idea what's wrong with my ledger. Did fava stop reporting syntax errors on the frontend?

**Comment by corani** [comment]
Duplicate of #329 that was fixed in beancount already. Not sure if it has
been released, or if the dependency has been updated in fava though.

Ping @aumayr

On Sun, Jul 3, 2016, 08:38 Xinruo Sun <notifications@github.com> wrote:

> option "operating_currency" "USD"
>
> 2016-01-01 open Expenses:Food
> 2016-01-01 open Assets:Cash
>
> 2016-06-05 * "Test"
>     Expenses:Food 100
>     Assets:Cash
>
> [E 160702 23:35:59 wsgi:355] 500 GET /beancount/income_statement/ (::1)
> 27.91ms
>
> Server 500 is generic internal error, I had no idea what's wrong with my
> ledger. Did fava stop reporting syntax errors on the frontend?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/aumayr/fava/issues/338>, or mute the thread
> <https://github.com/notifications/unsubscribe/AAdWB4Z5PCHPtlyOJXBgYLT6RFdf2f_Qks5qR1jNgaJpZM4JD0Kh>
> .
>
-- 

Best regards,
Daniël Bos

Your government is reading your email. Slow them down with encryption.

My public key: http://goo.gl/gms497 (4096 bit RSA, id EF2D5D91)
Fingerprint  : D8D0 9FBE F075 F709 7B52  2F73 326C 2123 EF2D 5D91

**Comment by aumayr** [comment]
@corani is absolutely right: This has been already addressed and fixed in beancount. But, as @corani speculated, there has not been an official PyPI release of beancount since this fix made it into the beancount-`master`, and no fava-release for that matter either. When the next fava-release is due (in the comming weeks), we will request a new release of beancount as well.

**Comment by yegle** [comment]
I was hit by this today.

```
[2016-11-25 08:36:24,663] ERROR in app: Exception on /beancount/income_statement/ [GET]
Traceback (most recent call last):
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/fava/application.py", line 245, in report
    return render_template('{}.html'.format(report_name))
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info,True)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/fava/templates/income_statement.html", line 3, in top-level template code
    {% import '_tree_table.html' as tree_table with context %}
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/fava/templates/_layout.html", line 53, in top-level template code
    <script type="application/json" id="chart-data">{{charts.chart_data|tojson|safe }}</script>
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/json.py", line 272, in tojson_filter
    return Markup(htmlsafe_dumps(obj, **kwargs))
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/json.py", line 186, in htmlsafe_dumps
    rv = dumps(obj, **kwargs) \
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/flask/json.py", line 126, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/__init__.py", line 397, in dumps
    **kw).encode(obj)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/encoder.py", line 293, in encode
    chunks = list(chunks)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/encoder.py", line 665, in _iterencode
    for chunk in _iterencode_list(o, _current_indent_level):
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/encoder.py", line 515, in _iterencode_list
    for chunk in chunks:
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/encoder.py", line 634, in _iterencode_dict
    for chunk in chunks:
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/encoder.py", line 515, in _iterencode_list
    for chunk in chunks:
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/encoder.py", line 634, in _iterencode_dict
    for chunk in chunks:
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/encoder.py", line 581, in _iterencode_dict
    k = _stringify_key(k)
  File "/Users/ych/.py3kvenv/lib/python3.5/site-packages/simplejson/encoder.py", line 551, in _stringify_key
    raise TypeError("key " + repr(key) + " is not a string")
TypeError: key <class 'beancount.core.number.MISSING'>is not a string
```
I'm using the latest release on PyPI and the code in the traceback doesn't involve beancount's code, so I assume this is still a bug in fava that's need to be fixed?

**Comment by blais** [comment]
I fixed a similar issue lately:
https://bitbucket.org/blais/beancount/commits/ccb1eb6a62f74288d1e4756f0bab8242c45edcf1

I was planning to address #145
https://bitbucket.org/blais/beancount/issues/145
before baking a new release.

Another factor is that the latest release's default booking method is still
the old one.
Baking a new release would make the new algorithm the default.
#145 points to an important issue which I need to fix.

Maybe I should fix #145 and make _that_ the official 2.0 and start 2.x
versioning from there on.

**Comment by blais** [comment]
Oh, and I meant to say: Can you update to head of default and check?
That should be gone I think.
I suspect invalid input (like #145 explains: the parser had to be loosened
for the new booking algorithm which is much more permissive in what you can
omit, its interpolation more powerful, and sometimes input that used to be
invalid under the old booking algorithm slips through without raising an
appropriate error).

If not, please isolate the offending transaction and file a bug with ti,
I'll get on it right away.
Anything that throws an exception like this is a P0 bug for me, very
important and urgent.
